### PR TITLE
Make RWKV-7 init match official RWKV-LM

### PR DESCRIPTION
**Pull request by johanwind**
The commit removes the following discrepancies with RWKV-LM:

1. Use _initialize_weights in RWKV7FeedForward. The init function for RWKV7FeedForward was implemented, but never called.
2. Initialize GroupNorms in RWKV7Attention. These were given default inits, while RWKV-LM uses a layer_idx-dependent scale.
3. nn.init.orthogonal_ instead of Tensor.uniform_ for RWKV7Attention's `r_proj`, `k_proj`, `v_proj` and RWKV7FeedForward's `key`.

After these changes, the initializations match. I verified it through monkeypatching torch.manual_seed(0) before torch random calls, which gives agreement between https://github.com/johanwind/wind_rwkv/blob/main/tests/train.py , https://github.com/BlinkDL/RWKV-LM/blob/main/RWKV-v7/train_temp/src/model.py , and https://github.com/BlinkDL/RWKV-LM/blob/main/RWKV-v5/src/model.py with
`args = SimpleNamespace(vocab_size = 65536 , n_layer = 12, n_embd = 768, head_size = 64, dim_ffn = 768*4)`,
and the updated FLA with 
`RWKV7ForCausalLM(RWKV7Config(hidden_size = 768, num_hidden_layers = 12, vocab_size = 65536, v_low_rank_dim = 32))`.

To further verify agreement, I did a training run with 1.5G tokens, by running https://github.com/johanwind/wind_rwkv/blob/main/tests/train.py with the model replaced by the corresponding FLA RWKV-7 (`RWKV7ForCausalLM(RWKV7Config(hidden_size = 768, num_hidden_layers = 12, vocab_size = 65536, v_low_rank_dim = 32))`) after the update. This gave a loss which is typical for the default RWKV. Before the updated inits, the results were slightly worse than the worst of the 8 runs I did with the default RWKV.

## Summary by CodeRabbit

* **Refactor**
  * Updated the initialization strategy for certain model layers to use orthogonal initialization, potentially improving model training stability.
  * Adjusted group normalization layer initialization for more consistent starting values.
  * Added conditional logic for weight initialization to enhance compatibility with different environments.

### CODE REVIEW

**Comment by sustcsonglin** [comment]
Thank you :)

**Comment by zhiyuan1i** [comment]
@johanwind  Thank you, but in latestest RWKV-LM, related codes here:

TMix:
https://github.com/BlinkDL/RWKV-LM/blob/main/RWKV-v7/train_temp/src/model.py#L157-L160

FFN:
https://github.com/BlinkDL/RWKV-LM/blob/main/RWKV-v7/train_temp/src/model.py#L216

**Comment by zhiyuan1i** [comment]
I think maybe the most important thing is the group learning rate for Lora components.

**Comment by johanwind** [comment]
> @johanwind Thank you, but in latestest RWKV-LM, related codes here:
> 
> TMix: https://github.com/BlinkDL/RWKV-LM/blob/main/RWKV-v7/train_temp/src/model.py#L157-L160
> 
> FFN: https://github.com/BlinkDL/RWKV-LM/blob/main/RWKV-v7/train_temp/src/model.py#L216

I believe those uniform_ initializations are overwritten in `generate_init_weight`, which I agree is confusing.

**Comment by zhiyuan1i** [comment]
> > @johanwind Thank you, but in latestest RWKV-LM, related codes here:
> > TMix: https://github.com/BlinkDL/RWKV-LM/blob/main/RWKV-v7/train_temp/src/model.py#L157-L160
> > FFN: https://github.com/BlinkDL/RWKV-LM/blob/main/RWKV-v7/train_temp/src/model.py#L216
> 
> I believe those uniform_ initializations are overwritten in `generate_init_weight`, which I agree is confusing.

I think you are right! Thank you.
